dali/util/file.cc

```diff
@@ -20,13 +20,22 @@
 #include "dali/util/std_file.h"
 #include "dali/util/uri.h"
 
+#if AWSSDK_ENABLED
+#include "dali/util/s3_client_manager.h"
+#include "dali/util/s3_file.h"
+#endif
+
 namespace dali {
 
 std::unique_ptr<FileStream> FileStream::Open(const std::string& uri, FileStream::Options opts,
                                              std::optional<size_t> size) {
   bool is_s3 = uri.rfind("s3://", 0) == 0;
   if (is_s3) {
+#if AWSSDK_ENABLED
+    return std::make_unique<S3FileStream>(S3ClientManager::Instance().client(), uri, size);
```

Comment:
Do we want to protest here if the user specifies incompatible flags,like use_mmap or use_odirect? S3FileStream constructor doesn't have access to the flags (because they're not supported), so this is the last chance.